### PR TITLE
.github/workflows/labels.yml: label PRs

### PR DESCRIPTION
**Pull request by zowoq**
###### Motivation for this change

- Easier to update when it's in the same repo.
- Labels will be accurate, e.g. `docker` won't be labelled with `documentation`.
- Doesn't require a redeploy of ofborg to pick up changes.

This will need to be merged after the labels are removed from `config.public.json` in ofborg and it is redeployed, I'll open a PR to do that if this one is accepted.

This will only affect labels that are listed here so the behaviour of manual and other `ofborg` labels (e.g. `rebuilds`, `security`, etc) will remain the same as it is currently.

### CODE REVIEW

File: .github/workflows/labels.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: "Label PR"
+
+on:
+  pull_request_target:
+
+jobs:
+  labels:
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'NixOS'
```
Comment (grahamc) [inline .github/workflows/labels.yml:9]: Does this mean it'll only labels PRs which are made from branches within the NixOS organization?

Comment (zowoq) [reply .github/workflows/labels.yml:9]: No, it just blocks the action from running in peoples forks.

File: .github/workflows/labels.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name: "Label PR"
+
+on:
+  pull_request_target:
```
Comment (marsam) [inline .github/workflows/labels.yml:4]: should we use just `pull_request`? https://securitylab.github.com/research/github-actions-preventing-pwn-requests actions/labels changed to `pull_request_target` to allow forked PRs also receive these labels, which doesn't seem the case for nixpkgs https://github.com/actions/labeler/issues/121#issuecomment-768239617

Comment (zowoq) [reply .github/workflows/labels.yml:4]: > changed to `pull_request_target` to allow forked PRs also receive these labels

We want PRs from forks to be labeled?

Comment (marsam) [reply .github/workflows/labels.yml:4]: oh, you are right! reading it again, I misunderstood their wording. Sorry for the noise.

**Comment by zowoq** [comment]
Forgot to mention that this will need to be backported and future changes to the list of labels would also need to be backported if we want the labels kept in sync for the release branch. I don't think that would be too much hassle?

**Comment by zowoq** [comment]
Opened a PR to remove these from `ofborg`: https://github.com/NixOS/ofborg/pull/553

**Comment by zowoq** [comment]
@cole-h @grahamc Can we move forward with this and https://github.com/NixOS/ofborg/pull/553?

**Comment by cole-h** [comment]
Sure. However, I won't have time to monitor an ofborg deploy until this weekend (and after work).

/me sets a reminder.

**Comment by cole-h** [comment]
Deployed, and it works! Thanks @zowoq :)

(Test PR: https://github.com/NixOS/nixpkgs/pull/117084)

**Pull request merged**